bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/IExpandableNode.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Advantest Germany GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * 				Raghunandana Murthappa
+ * 				Andrey Loskutov (loskutov@gmx.de)
+ *******************************************************************************/
+package org.eclipse.jface.viewers;
+
+/**
+ * The expandable placeholder element to be used for viewer items that represent
+ * an expandable tree or table element.
+ * <p>
+ * The idea of {@link IExpandableNode} is to allow viewers to show only some
+ * subset of children that would otherwise all appear below parent element. The
+ * main purpose of this is to prevent UI freezes with viewers that can offer lot
+ * of elements but can't efficiently handle such amount using SWT.
+ * <p>
+ * The node consists of a parent element, list of all children of this parent
+ * and the offset to which child elements are supposed to be created and shown
+ * in the viewer.
+ *
+ * @since 3.31
+ */
+public interface IExpandableNode {
```

Comment:
I believe for now we should add `noimplement` annotation here, as we don't plan yet to support third party expandable nodes

Reply:
Does it really have to be a public interface? Can't it be internal?